### PR TITLE
add more examples

### PR DESCRIPTION
**Pull request by garbados**
## Proposed Release Notes

- Added more API usage examples.
- Added a README to the `examples/` folder discussing how to use the examples.

## Links

- Closes https://github.com/newrelic/node-newrelic/issues/441

## Details

As of this writing, this PR is a WIP. More examples are forthcoming.

### CODE REVIEW

**Comment by garbados** [comment]
- [x] newrelic.recordCustomEvent(eventType, attributes)
- [x] newrelic.addCustomAttribute(name, value)
- [ ] newrelic.addCustomAttributes(attributes)
- [ ] newrelic.addCustomSpanAttribute(name, value)
- [ ] newrelic.addCustomSpanAttributes(attributes)
- [ ] transactionHandle.acceptDistributedTraceHeaders
- [ ] transactionHandle.insertDistributedTraceHeaders
- [ ] transactionHandle.createDistributedTracePayload()
- [ ] transactionHandle.acceptDistributedTracePayload

**Comment by garbados** [comment]
Of note:

> This method (acceptDistributedTraceHeaders) replaces the deprecated (and now removed as of version 7.0.0) acceptDistributedTracePayload method, which only handles New Relic distributed trace payloads.

Should I still include examples for the deprecated createDistributedTracePayload and acceptDistributedTracePayload methods?

**Comment by michaelgoin** [comment]
@garbados I would not. At this point they have been fully removed and I expect the in-repo examples to be representative of the current state.

**Comment by garbados** [comment]
> @garbados I would not. At this point they have been fully removed and I expect the in-repo examples to be representative of the current state.

Great! Then this PR is ready for review.

File: examples/api/background-transactions/example-acceptDistributedTraceHeaders.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+Accepted header keys include:
+
+- `traceparent`: A W3C Trace Context header.
+- `newrelic`: New Relic distributed trace header.
+*/
+
+newrelic.startBackgroundTransaction('myCustomTransaction', function handle() {
+  const transaction = newrelic.getTransaction()
+  transaction.acceptDistributedTraceHeaders('Kafka', { traceparent: 'payload' })
```
Comment (carlo-808) [inline examples/api/background-transactions/example-acceptDistributedTraceHeaders.js:37]: I think we already have an example of this here: [examples/api/distributed-tracing/example1-background.js](https://github.com/newrelic/node-newrelic/blob/main/examples/api/distributed-tracing/example1-background.js)
but I like the notes you added and that example is not too great either. I think you can improve this kafka example by showing how to generate the DT/tracecontext headers on a kafka producer and retrieve them on a kafka consumer. I'll send you the example over slack.

File: examples/api/background-transactions/example-insertDistributedTraceHeaders.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
```
Comment (carlo-808) [inline examples/api/background-transactions/example-insertDistributedTraceHeaders.js:1]: May want to combine this with the acceptDistributedTraceHeaders example.

**Review by carlo-808 — COMMENTED** [review]
I added some notes on the Kafka example. Other examples look great!

File: examples/api/background-transactions/example-acceptDistributedTraceHeaders.js (hunk 1 above)
Comment (garbados) [reply examples/api/background-transactions/example-acceptDistributedTraceHeaders.js]: After reviewing the examples you sent, I've gone and reconciled the examples I added with that pre-existing example, significantly expanding its docs. Let me know what you think!

Comment (carlo-808) [reply examples/api/background-transactions/example-acceptDistributedTraceHeaders.js]: Looks good! Thanks for updating.

**Review by carlo-808 — APPROVED** [review]
Great addition of examples with added context in comments.

**Pull request merged**